central/alert/datastore/internal/search/searcher_impl.go

```diff
@@ -55,6 +50,18 @@ func (ds *searcherImpl) SearchAlerts(ctx context.Context, q *v1.Query) ([]*v1.Se
 
 // SearchListAlerts retrieves list alerts from the indexer and storage
 func (ds *searcherImpl) SearchListAlerts(ctx context.Context, q *v1.Query) ([]*storage.ListAlert, error) {
+	if env.PostgresDatastoreEnabled.BooleanSetting() {
```

Comment:
Should we add `ensurePaginated` to getAll/List type of functions in our code? The check will ensure we are not pulling crazy data into memory. (Maybe we can do it as part of search productization epic)

Reply:
I think I'd rather alert on the number of rows returned. Sometimes we have large queries that actually use that without pagination. For example, the alerts in alertmanager don't use pagination. We'd have to offer something like next page to make it really easy